wgpu-core/src/device/resource.rs

```diff
@@ -590,6 +629,10 @@ impl Device {
         };
         let buffer = unsafe { self.raw().create_buffer(&hal_desc) }.map_err(DeviceError::from)?;
 
+        #[cfg(feature = "indirect-validation")]
+        let raw_indirect_validation_bind_group =
+            self.create_indirect_validation_bind_group(buffer.as_ref(), desc.size, desc.usage)?;
+
```

Comment:
thought(non-blocking): It seems strange to "unconditionally" create this bind group. I'm not immediately sure how this could be improved, though; the only things I can think of to clear this up would be to effectively inline the helper.

Reply:
It can't be avoided and it's happening at buffer creation time so it shouldn't be surprising from a perf perspective.

The helper is used in 2 places (this being one of them), I'm not sure why it would be better to inline it?

Reply:
Sorry, my feedback is not intended to comment on how semantics need to change; I think the code is doing the right thing. I think the way that the code is communicating ("let's 'create' something, but I under-the-hood might not actually create anything") is disorienting to somebody on both sides of a black box, but it's extremely minor. I agree, though, that I don't think there's a meaningful action to take here WRT clarifying; I can't think of something better to do.

I feel heard after expressing the above, so I'm going to `Resolve`.